app/models/user.rb

```diff
@@ -1,4 +1,18 @@
 class User < ApplicationRecord
   attr_accessor :display_name, :admin, :roles
   validates :login, presence: true
+
+  def self.from_jwt(data = { })
+    id, login = data&.values_at 'id', 'login'
+    return unless id && login
+
+    User.where(id: id, login: login).first_or_create do |user|
+      user.display_name = data['dname']
+      user.admin = data['admin'] == true # explicit
```

Comment:
Explicitly checking the JWT for the encoded admin flag.

Reply:
```suggestion
      # explicit set the internal admin flag if the user is an admin
      user.admin = data['admin'] == true 
```
What happens longer term, where say an admin leaves and loses admin status in main API? they'll still have admin rights on this system, right?

Reply:
No, because this system would be parsing their new JWT forwarded by the front end and it wouldn't have the admin flag set. This is per request.

Reply:
Does `first_or_create` run the block on each request or only when the record is first created?

Reply:
For the record (and in case this was a genuine question), the block isn't run if the record isn't created. Added `.tap` to the line to always run the block.